checks.d/docker_daemon.py

```diff
@@ -432,13 +436,24 @@ def refresh_ecs_tags(self):
         ip = ecs_config.get('NetworkSettings', {}).get('IPAddress')
         ports = ecs_config.get('NetworkSettings', {}).get('Ports')
         port = ports.keys()[0].split('/')[0] if ports else None
+        if not ip and DockerUtil.is_dockerized():
+            if self.docker_gateway:
+                ip = self.docker_gateway
+                port = ECS_INTROSPECT_DEFAULT_PORT
+        elif not ip:
+            ip = "localhost"
+            port = ECS_INTROSPECT_DEFAULT_PORT
+
         ecs_tags = {}
-        if ip and port:
-            tasks = requests.get('http://%s:%s/v1/tasks' % (ip, port)).json()
-            for task in tasks.get('Tasks', []):
-                for container in task.get('Containers', []):
-                    tags = ['task_name:%s' % task['Family'], 'task_version:%s' % task['Version']]
-                    ecs_tags[container['DockerId']] = tags
+        try:
+            if ip and port:
+                tasks = requests.get('http://%s:%s/v1/tasks' % (ip, port)).json()
+                for task in tasks.get('Tasks', []):
+                    for container in task.get('Containers', []):
+                        tags = ['task_name:%s' % task['Family'], 'task_version:%s' % task['Version']]
+                        ecs_tags[container['DockerId']] = tags
+        except requests.exceptions.HTTPError as e:
```

Comment:
Is it possible to get a requests.exceptions.ConnectionError here? I feel like this could happen if the ecs-agent crashes momentarily and we query the endpoint at this time, or if the default gateway doesn't resolve to it, which is the case in some setups where a network overlay tool is in use.